policy/main.go

```diff
@@ -97,48 +100,128 @@ func main() {
 			log.Println(err)
 			os.Exit(1)
 		}
+		accessToken, err = kmsDecrypt(accessToken)
+		if err != nil {
+			log.Println(err)
+			os.Exit(1)
+		}
+		refreshToken, err = kmsDecrypt(refreshToken)
+		if err != nil {
+			log.Println(err)
+			os.Exit(1)
+		}
 	}
 
 	vcertConnector, err = getConnection(
 		os.Getenv("TPPURL"),
 		os.Getenv("TPPUSER"),
 		password,
-		os.Getenv("CLOUDURL"),
+		accessToken,
+		refreshToken,
 		apiKey,
 		os.Getenv("TRUST_BUNDLE"),
 	)
 	if err != nil {
 		log.Println(err)
 		os.Exit(1)
 	}
+
 	lambda.Start(HandleRequest)
 }
 
-func getConnection(tppUrl, tppUser, tppPassword, cloudUrl, cloudKey, trustBundle string) (endpoint.Connector, error) {
+func getConnection(tppUrl, tppUser, tppPassword, accessToken, refreshToken, apiKey, trustBundle string) (endpoint.Connector, error) {
 	log.Println("Getting Venafi connection")
 	var config vcert.Config
 	if tppUrl != "" && tppUser != "" && tppPassword != "" {
 		config = vcert.Config{
 			ConnectorType: endpoint.ConnectorTypeTPP,
 			BaseUrl:       tppUrl,
-			Credentials:   &endpoint.Authentication{User: tppUser, Password: tppPassword},
+			Credentials: &endpoint.Authentication{
+				User:     tppUser,
+				Password: tppPassword,
+			},
 		}
-		if trustBundle != "" {
-			buf, err := base64.StdEncoding.DecodeString(trustBundle)
-			if err != nil {
-				log.Printf("Can`t read trust bundle from file %s: %v\n", trustBundle, err)
-				return nil, err
-			}
-			config.ConnectionTrust = string(buf)
+
+	} else if tppUrl != "" && (accessToken != "" || refreshToken != "") {
+		config = vcert.Config{
+			ConnectorType: endpoint.ConnectorTypeTPP,
+			BaseUrl:       tppUrl,
+			Credentials: &endpoint.Authentication{
+				AccessToken:  accessToken,
+				RefreshToken: refreshToken,
+				ClientId:     ClientId,
+			},
 		}
-	} else if cloudKey != "" {
+	} else if apiKey != "" {
 		config = vcert.Config{
 			ConnectorType: endpoint.ConnectorTypeCloud,
-			Credentials:   &endpoint.Authentication{APIKey: cloudKey},
-			BaseUrl:       cloudUrl,
+			Credentials: &endpoint.Authentication{
+				APIKey: apiKey,
```

Comment:
Not Sure if is github editor or you may need to format this part of the code,

Reply:
APIKey is a field of the Credentials struct, thats why it has more indentation.